druid/src/event.rs

```diff
@@ -137,29 +137,30 @@ pub enum Event {
 #[derive(Debug, Clone)]
 pub enum LifeCycle {
     /// Sent to a `Widget` when it is added to the widget tree. This should be
-    /// the first message that each widget receives.
+    /// the first message that each widget receives. This should only be sent once.
+    /// If any new children are added a RouteWidgedAdded event will be sent instead.
```

Comment:
I appreciate the effort to try and keep the docs up to date; that's an important thing to do.

That said, I'm not 100% sure what the right docs are, here. My basic feeling is that these docs might be slightly outdated, since registering children is handled entirely in the `WidgetPod`.

So I think the easiest thing to do is to leave these docs as-is, and we can open an issue about them being slightly out of date. For `RouteWidgetAdded`, I would include minimal docs, similar to `RouteFocusChanged`; basically I would say something like, "Internal: used by the framework to route the `WidgetAdded` event."

We might also want to consider adding `#[doc(hidden)]` to the various 'route' events, but that's also a separate concern.